pkg/virt-controller/watch/vmi.go

```diff
@@ -2127,10 +2127,14 @@ func (c *VMIController) updateVolumeStatus(vmi *virtv1.VirtualMachineInstance, v
 }
 
 func (c *VMIController) getFilesystemOverhead(pvc *k8sv1.PersistentVolumeClaim) (cdiv1.Percent, error) {
-	_, cdiExists, _ := c.cdiInformer.GetStore().GetByKey("cdi")
-	if !cdiExists {
+	// To avoid conflicts, we only allow having one CDI instance
+	if cdiInstances := len(c.cdiInformer.GetStore().List()); cdiInstances != 1 {
+		if cdiInstances > 1 {
+			return "0", fmt.Errorf("Detected more than one CDI instance: %d", cdiInstances)
```

Comment:
only nitpick is that we have a named error for `failedToFindCdi` and not one for multipleCdi. I would do either both named or neither named to be consistent.

Reply:
I agree but I see a minor problem there: `failedToFindCdi` is explicitly used in the error handling, so I think it makes sense that we use a named error just in that case. I can also create a new error for the other case, like `multipleCdiInstances`, but it wouldn't be used in the error-handling nor include the number of instances, which I like.

Reply:
Yeah, I agree. I'll change that :+1: